src/main/java/org/apache/sysml/hops/AggUnaryOp.java

```diff
@@ -329,8 +329,15 @@ public boolean allowsAllExecTypes()
 
 	@Override
 	protected double computeOutputMemEstimate( long dim1, long dim2, long nnz )
-	{		
-		double sparsity = OptimizerUtils.getSparsity(dim1, dim2, nnz);
+	{
+		double sparsity = -1;
+		if (DMLScript.USE_ACCELERATOR) {
+			// The GPU version (for the time being) only does dense outputs
+			sparsity = 1.0;
+		} else {
+			sparsity = OptimizerUtils.getSparsity(dim1, dim2, nnz);
+		}
```

Comment:
I was referring to a scenario where the subsequent (data-dependent) operation cannot be compiled to GPU - in this case this wrong output estimate would negatively affect the subsequent operation (e.g., SPARK instead of CP). You have two options here: (1) either check the parent (for applicable GPU operations), or (2) leave it as proposed as we'll anyway clean this up once there is proper support for dense and sparse as well as separate memory budgets and estimates. I leave it up to you.